components/nexus-repository/src/test/java/org/sonatype/nexus/repository/storage/StorageFacetImplIT.java

```diff
@@ -561,4 +604,160 @@ public void duplicateAssetName() throws Exception {
+            .group("myGroup")
+            .version("0.9")
+            .name("myComponent");
+        tx.saveComponent(component);
+        if (true) {
+          throw new MidTransactionException();
+        }
+      }
+    }
+    catch (MidTransactionException ignored) {
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(nullValue()));
+    }
+  }
+
+  @Test
+  public void transactionContentIsSaved() throws Exception {
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat)
+          .group("myGroup")
+          .version("0.9")
+          .name("myComponent");
+      tx.saveComponent(component);
+      tx.commit();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Iterable<Component> components = tx.browseComponents(tx.getBucket());
+      for (Component c : components) {
+        System.err.println(c);
+      }
+
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(notNullValue()));
+      assertThat(component.group(), is("myGroup"));
+    }
+  }
+
+  @Test
+  public void entityIdCanBeUsedInLaterTransactions() throws Exception {
+    EntityId componentId = null;
+    EntityId assetId = null;
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat).name("component");
+      tx.saveComponent(component);
+
+      componentId = id(component);
+
+      tx.commit();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponent(componentId, tx.getBucket());
+      assertThat("component", component, is(notNullValue()));
+      assertThat(component.name(), is("component"));
+    }
+  }
+
+  @Test
+  public void multipleEntityIdsCanBeUsedInLaterTransactions() throws Exception {
+    EntityId componentId = null;
+    EntityId assetId = null;
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat).name("component");
+      tx.saveComponent(component);
+
+      final Asset asset = tx.createAsset(tx.getBucket(), component).name("hello");
+      tx.saveAsset(asset);
+
+      tx.commit();
+      componentId = id(component);
+      assetId = id(asset);
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponent(componentId, tx.getBucket());
+      assertThat("component", component, is(notNullValue()));
+      assertThat(component.name(), is("component"));
+
+      final Asset asset = tx.findAsset(assetId, tx.getBucket());
+      assertThat("asset", asset, is(notNullValue()));
+      assertThat(asset.name(), is("hello"));
+    }
+  }
+
+  private static class MidTransactionException
```

Comment:
really no need for custom exception here, is there? seems like overkill, even if it's just a couple of lines